[CR] Fix incorrect damage messages from vault zombie attacks

mightyagrippa opened the pull request:
Addresses https://github.com/CleverRaven/Cataclysm-DDA/issues/13194, which is still present. Player messaging is changed to only list damage if attack actually gets through armor. 

Irradiated wanderer balance is unchanged. Didn't do damage through armor before, did knockdown randomly through armor, still does the same thing. It makes sense that even a walking ball of kevlar can be knocked down.

Charred nightmare balance is changed. It applied head bleed through armor regardless of damage before, now it requires damage > 0 to apply the head bleed. Not 100% sure on this but seemed like the right thing to do.

Coolthulhu commented on src/monattack.cpp:
```diff
@@ -3963,16 +3963,20 @@ bool mattack::lunge(monster *z)
+        target->add_msg_if_player(m_bad, _("Your %1$s is battered for %2$d damage!"), body_part_name(hit).c_str(), dam);
+    } else {
+        target->add_msg_player_or_npc( _("The %1$s lunges at your %2$s, but glances off your armor!"),
+                                    _("The %1$s lunges at <npcname>'s %2$s, but glances off armor!"),
```
Glancing off suggests the effect is totally negated, but downing can still happen. Not sure how to say that better, though.
Lunging is already mentioned above. You could remove that message and integrate it here or keep it as 2 messages "lunges into" and then "but fails to deal damage". The former (one message per attack) would be preferable as most attacks look like that.
When a npc/monster avoids all damage a msg is printed, but no message is printed if npc/monster takes damage from the attack. It's OK to reveal how hard was a NPC/monster hit.

mightyagrippa commented:
@Coolthulhu, looking over monattack again it seems like very few attacks give explicit damage numbers. Should those be removed here for player and NPC damage notices?

One possibility for the damage phrasing would be "The irradiated wanderer lunges at your torso, but your armor prevents injury." That scans better with the ensuing "You're knocked to the floor" effect notification.

Coolthulhu commented:
> Should those be removed here for player and NPC damage notices?

No need, it's fine - it doesn't really hide much data, as you can just check it later (with some loss of precision) with 'x' or 'V'.
Most of the ones that don't show it are old functions that didn't have access to damage numbers at the time they were written.

mightyagrippa commented:
Ok, building and testing for desired behavior now. This is my first contribution so please speak up if I'm doing something wrong or out of order. And thanks for the comments.

mightyagrippa commented:
Agreed on the specificity, changed that. I also changed the m_info to m_warning in the attacks in question. Testing showed everything behaved as expected, so the only other change was to make all versions of the longswipe ranged version display "thrust a claw" for consistency.

kevingranade commented:
Not a must for this PR, but I favor removing damage numbers, you can see the effects, but eventually HP are going away entirely, so the damage shouldn't be displayed.

Coolthulhu commented:
I'd wait with removing damage numbers until the damage is removed.
For now hiding numbers just makes it more tedious to approximate critter's strength and hides things that would be clearly visible to the characters in game. Things like defenders getting barely scratched vs. reeling in pain, it's just "it did go through armor" or "it didn't" (and in worst case just "it did connect" or "target dodged").

mightyagrippa commented:
Even with hp gone I'd want some indication of how damaging a hit was ("scratched your throat", "slashed your throat", "maimed your throat") so I'm fine with leaving hp damage indicators in until a replacement is in the works.

I'm new to git, aside from discussion are there any other steps needed on my end for this to be ready?

Coolthulhu commented:
> I'm new to git, aside from discussion are there any other steps needed on my end for this to be ready?

Unless testing shows up some unexpected bugs, it's all done on your end.
